frontend/src/components/taxons/EditTaxon.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button, Form, Input, Space } from "antd";
+import React, { useCallback, useState } from "react";
+import { requiredRules } from "../../constants";
+import { useAppDispatch, useAppSelector } from "../../hooks";
+import { selectAppInitialzed, selectTaxonsByID } from "../../selectors";
+import { useNavigate, useParams } from "react-router-dom";
+import { add, update } from "../../modules/taxons/actions";
+import AppPageHeader from "../AppPageHeader";
+import PageContent from "../PageContent";
+
+export interface Taxon {
+    id?: number | null
+    ncbi_id: string
+    name: string
+}
+export interface EditTaxonProps {
+    taxon: Taxon
+}
+
+export const AddTaxonRoute = () => {
+    const emptyTaxon = { ncbi_id: "", name: "" }
+    const appInitialzed = useAppSelector(selectAppInitialzed)
+
+    return appInitialzed ? <EditTaxon taxon={emptyTaxon} /> : null
+
+}
+export const EditTaxonRoute = () => {
+    const taxons = useAppSelector(selectTaxonsByID)
+    const { id } = useParams();
+    const appInitialzed = useAppSelector(selectAppInitialzed)
+
+    return appInitialzed ? <EditTaxon taxon={taxons[id ?? -1]} /> : null
+
+}
+
+const EditTaxon = ({ taxon }: EditTaxonProps) => {
+    const { Item } = Form
+    const [formErrors, setFormErrors] = useState({})
+    const [form] = Form.useForm()
+    const isAdding = taxon.id === undefined
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch()
+
+    const props = (name: string) =>
+        !formErrors[name] ? { name } : {
+            name,
+            hasFeedback: true,
+            // validateStatus: "error", 
+            help: formErrors[name],
+        }
+    const onFinish = () => {
+        const new_taxon: Taxon = taxon;
+        const fieldValues = form.getFieldsValue();
+        Object.keys(fieldValues).forEach((fieldName: string) => {
```

Comment:
I'm not crazy about using Object.keys here. It's defeating typescript's type checking. Since there are only a few fields, it might be better to just assign the id, ncbi_id and name properties explicitly.